### ISSUE TITLE
Showing Telegram's notifications on the lock screen

### ISSUE THREAD

**Issue by zolagonano** [i3/i3lock#333]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

i3lock shows telegram's notifications(and probably other apps as well) while its locked, which might contain sensitive information.

## Expected Behavior

Not showing anything from outside lock screen.

## Reproduction Instructions

1. open telegram desktop
2. lock your device
3. send a message to your telegram account, and you'll get the message on the lock screen

## Environment
Output of `i3lock --version`:
<pre>
i3lock version:  2.14.1 
</pre>

**Comment by stapelberg** [comment]
Are you using a compositor like `xcompmgr`, `compton`, `picom` or others? If so, try turning it off.

**Comment by zolagonano** [comment]
> Are you using a compositor like `xcompmgr`, `compton`, `picom` or others? If so, try turning it off.

Yeah i do.
I tried turning them off and it fixed the problem.
but i still need my compositor, is there any fix for that? without having to remove the compositor?

**Comment by stapelberg** [comment]
I don’t recall if there are better workarounds, but you can wrap i3lock in a script that kills the compositor, invokes i3lock, then restarts the compositor.

**Comment by kolayne** [comment]
Killing `picom` didn't work for me. Running Telegram version 4.7.1, i3lock: version 2.14.1

**Comment by ssidashov** [comment]
I have the same problem, and I have no picom or any other compositor installed. Telegram messages are showed on lock screen. Btw, betterlockscreen has the same problem. 
i3lock: version 2.13.c.4

**Comment by stapelberg** [comment]
I cannot reproduce this problem without a compositor.

I tried it with Telegram’s native notifications and custom notifications. Neither show up on i3lock when not running a compositor.

As soon as I start a compositor, both mechanisms show up.

To double-check whether you’re running a compositor: run `notify-send foo` via SSH. If the message is visible on i3lock, you’re running a compositor.

> i3lock: version 2.13.c.4

This is not a known version number. We have released i3lock 2.13, followed by 2.14. I don’t know what the “c.4” stands for. I suggest you try it with upstream i3lock, without modifications.

**Comment by ssidashov** [comment]
> This is not a known version number. We have released i3lock 2.13, followed by 2.14. I don’t know what the “c.4” stands for. I suggest you try it with upstream i3lock, without modifications.
Yes, i'm sorry, this is a version of fork i3lock-color (https://github.com/Raymo111/i3lock-color), which is betterlockscreen dependency  in Arch.

**Comment by ssidashov** [comment]
> This is not a known version number. We have released i3lock 2.13, followed by 2.14. I don’t know what the “c.4” stands for. I suggest you try it with upstream i3lock, without modifications.

Yes, i'm sorry, this is a version of fork i3lock-color (https://github.com/Raymo111/i3lock-color), which is betterlockscreen dependency  in Arch.

**Comment by micronull** [comment]
The problem is reproduced on version 2.14.1
```sh
i3lock & sleep 1 && notify-send foo
```

**Comment by tycho-kirchner** [comment]
While I prefer using i3lock I wanted to point out that gnome-screensaver is able to hide the notifications, even if a compositor is running. I don't know, how, but it might be worth looking into.
~~~
gnome-screensaver-command --lock & sleep 1; notify-send foo
~~~

**Comment by zolagonano** [comment]
aah, this works well thank you. but all this long, i guess the fix to this problem was removing the program itself

-------- Original Message --------

**Comment by tycho-kirchner** [comment]
> the fix to this problem was removing the program itself

Uh, which program did u remove 😅?

**Comment by zolagonano** [comment]
i3locks. switched to gnome screensaver, although im not sure if it was an i3locks problem or a compositor problem. but anyways, something had to be removed.

-------- Original Message --------

**Comment by tycho-kirchner** [comment]
Ah ok, actually I meant to look into the source of gnome-screensaver, see how it's done there and implement that in i3lock. But that's for sure a bit of work to do (;

**Comment by zolagonano** [comment]
that'd be super nice tbh, i3lock was much simpler and less bloated.

-------- Original Message --------
